src/Scanner.js

```diff
@@ -148,13 +139,39 @@ class DefaultScanner {
         } else {
             const inputs = context.querySelectorAll(FORM_INPUTS_SELECTOR)
             if (inputs.length > this.options.maxInputsPerPage) {
+                this.stopScanner('Too many input fields in the given context, stop scanning', context)
                 return this
             }
             inputs.forEach((input) => this.addInput(input))
         }
         return this
     }
 
+    /**
+     * Stops scanning, switches off the mutation observer and clears all forms
+     * @param {string} reason
+     * @param {...any} rest
+     */
+    stopScanner (reason, ...rest) {
```

Comment:
I've centralized this stop method here so we use it consistently and can log why the scanner has stopped. The main driver of the change is to ensure we also `disconnect` the `MutationObserver` when we stop the scans, which wasn't happening before.